From dd1f4ef0b01724b28860ffb39ae14b2b6bdad4d7 Mon Sep 17 00:00:00 2001
From: skandigraun <skandigraun@gmail.com>
Date: Tue, 30 May 2023 22:17:51 +0200
Subject: [PATCH] update firefox

This patch changes the Cargo.toml files to use local crates instead of fetching
custom ones from github - the crates are the same, but they are fetched in
do_fetch task instead of allowing cargo to fetch them during compiling, when
there is no network connection.

Upstream-Status: Inappropriate [oe specific]
---
 .cargo/config.in                    |  2 +-
 Cargo.toml                          | 92 +++++++++++++++++++++--------
 third_party/rust/suggest/Cargo.toml |  8 +--
 3 files changed, 69 insertions(+), 33 deletions(-)

--- ./.cargo/config.toml.in	2025-04-28 15:44:40.964583525 +0200
+++ ./.cargo/config.toml.in	2025-04-28 15:45:03.364366743 +0200
@@ -123,7 +123,7 @@
 # cargo would fail.
 #ifndef REPLACE_NAME
 [source.vendored-sources]
-directory = "third_party/rust"
+directory = "../../cargo_home/bitbake"
 #endif
 
 # Thankfully, @REPLACE_NAME@ is unlikely to be a legitimate source, so
--- ./Cargo.toml	2025-08-19 22:44:26.085999031 +0200
+++ ./Cargo.toml	2025-08-19 22:47:29.357875065 +0200
@@ -222,20 +222,20 @@
 unicode-width = { path = "build/rust/unicode-width" }
 
 # To-be-published changes.
-unicode-bidi = { git = "https://github.com/servo/unicode-bidi", rev = "ca612daf1c08c53abe07327cb3e6ef6e0a760f0c" }
-nss-gk-api = { git = "https://github.com/beurdouche/nss-gk-api", rev = "e48a946811ffd64abc78de3ee284957d8d1c0d63" }
-cssparser = { git = "https://github.com/servo/rust-cssparser", rev = "958a3f098acb92ddacdce18a7ef2c4a87ac3326f" }
+unicode-bidi = { path = "../unicode-bidi" }
+nss-gk-api = { path = "../nss-gk-api" }
+cssparser = { path = "../cssparser" }
 
 # Other overrides
-any_all_workaround = { git = "https://github.com/hsivonen/any_all_workaround", rev = "7fb1b7034c9f172aade21ee1c8554e8d8a48af80" }
-chardetng = { git = "https://github.com/hsivonen/chardetng", rev = "3484d3e3ebdc8931493aa5df4d7ee9360a90e76b" }
-chardetng_c = { git = "https://github.com/hsivonen/chardetng_c", rev = "ed8a4c6f900a90d4dbc1d64b856e61490a1c3570" }
-cose = { git = "https://github.com/franziskuskiefer/cose-rust", rev = "43c22248d136c8b38fe42ea709d08da6355cf04b" }
+any_all_workaround = { path = "../any-all-workaround" }
+chardetng = { path = "../chardetng" }
+chardetng_c = { path = "../chardetng_c" }
+cose = { path = "../cose-rust" }
 firefox-on-glean = { path = "toolkit/components/glean/api" }
 icu_capi = { path = "intl/icu_capi" }
 icu_segmenter_data = { path = "intl/icu_segmenter_data" }
 libudev-sys = { path = "dom/webauthn/libudev-sys" }
-midir = { git = "https://github.com/mozilla/midir.git", rev = "37ad39de3382c2aad0758dab49ba545331a2257d" }
+midir = { path = "../midir" }
 # Allow webrender to have a versioned dependency on the older crate on crates.io
 # in order to build standalone.
 malloc_size_of_derive = { path = "xpcom/rust/malloc_size_of_derive" }
@@ -243,22 +243,68 @@
 wr_malloc_size_of = { path = "gfx/wr/wr_malloc_size_of" }
 
 # objc 0.2.7 + fa7ca43b862861dd1cd000d7ad01e6e0266cda13
-objc = { git = "https://github.com/glandium/rust-objc", rev = "4de89f5aa9851ceca4d40e7ac1e2759410c04324" }
+objc = { path = "../objc" }
 
 # application-services overrides to make updating them all simpler.
-context_id = { git = "https://github.com/mozilla/application-services", rev = "b5175a2a2458834c9fdc0468d851658b0657f95b" }
-filter_adult = { git = "https://github.com/mozilla/application-services", rev = "b5175a2a2458834c9fdc0468d851658b0657f95b" }
-interrupt-support = { git = "https://github.com/mozilla/application-services", rev = "b5175a2a2458834c9fdc0468d851658b0657f95b" }
-relevancy = { git = "https://github.com/mozilla/application-services", rev = "b5175a2a2458834c9fdc0468d851658b0657f95b" }
-search = { git = "https://github.com/mozilla/application-services", rev = "b5175a2a2458834c9fdc0468d851658b0657f95b" }
-sql-support = { git = "https://github.com/mozilla/application-services", rev = "b5175a2a2458834c9fdc0468d851658b0657f95b" }
-suggest = { git = "https://github.com/mozilla/application-services", rev = "b5175a2a2458834c9fdc0468d851658b0657f95b" }
-sync15 = { git = "https://github.com/mozilla/application-services", rev = "b5175a2a2458834c9fdc0468d851658b0657f95b" }
-tabs = { git = "https://github.com/mozilla/application-services", rev = "b5175a2a2458834c9fdc0468d851658b0657f95b" }
-viaduct = { git = "https://github.com/mozilla/application-services", rev = "b5175a2a2458834c9fdc0468d851658b0657f95b" }
-webext-storage = { git = "https://github.com/mozilla/application-services", rev = "b5175a2a2458834c9fdc0468d851658b0657f95b" }
+context_id = { path = "../application-services/components/context_id", version = "0.1.0" }
+filter_adult = { path = "../application-services/components/filter_adult", version = "0.1.0" }
+interrupt-support = { path = "../application-services/components/support/interrupt", version = "0.1.0" }
+relevancy = { path = "../application-services/components/relevancy", version = "0.1.0"}
+search = { path = "../application-services/components/search", version = "0.1.0" }
+sql-support = { path = "../application-services/components/support/sql", version = "0.1.0" }
+suggest = { path = "../application-services/components/suggest", version = "0.1.0" }
+sync15 = { path = "../application-services/components/sync15", version = "0.1.0" }
+tabs = { path = "../application-services/components/tabs", version = "0.1.0" }
+error-support = { path = "../application-services/components/support/error", version = "0.1.0" }
+viaduct = { path = "../application-services/components/viaduct", version = "0.1.0" }
+webext-storage = { path = "../application-services/components/webext-storage", version = "0.1.0" }
 
 allocator-api2 = { path = "third_party/rust/allocator-api2" }
 
 # Patched version of zip 2.4.2 to allow for reading omnijars.
 zip = { path = "third_party/rust/zip" }
+
+[patch."https://github.com/FirefoxGraphics/aa-stroke"]
+aa-stroke = { path = "../aa-stroke"}
+
+[patch."https://github.com/mozilla/audioipc"]
+audioipc2-client = { path = "../audioipc/client" }
+audioipc2-server = { path = "../audioipc/server" }
+
+[patch."https://github.com/mozilla/cubeb-coreaudio-rs"]
+cubeb-coreaudio = { path = "../cubeb-coreaudio" }
+
+[patch."https://github.com/mozilla/cubeb-pulse-rs"]
+cubeb-pulse = { path = "../cubeb-pulse" }
+
+[patch."https://github.com/jfkthame/mapped_hyph.git"]
+mapped_hyph = { path = "../mapped_hyph" }
+
+[patch."https://github.com/mozilla/mp4parse-rust"]
+mp4parse_capi = { path = "../mp4parse/mp4parse_capi", version = "0.17.0" }
+
+[patch."https://github.com/FirefoxGraphics/wpf-gpu-raster"]
+wpf-gpu-raster = { path = "../wpf-gpu-raster" }
+
+[patch."https://github.com/gfx-rs/wgpu"]
+wgpu-core = { path = "../wgpu/wgpu-core" }
+wgpu-types = { path = "../wgpu/wgpu-types" }
+wgpu-hal = { path = "../wgpu/wgpu-hal" }
+wgpu-core-deps-windows-linux-android = { path = "../wgpu/wgpu-core/platform-deps/windows-linux-android" }
+
+[patch."https://github.com/mozilla/neqo"]
+neqo-bin = { path = "../neqo/neqo-bin", version = "0.14.1" }
+neqo-http3 = { path = "../neqo/neqo-http3", version = "0.14.1" }
+neqo-common = { path = "../neqo/neqo-common", version = "0.14.1" }
+neqo-transport = { path = "../neqo/neqo-transport", version = "0.14.1" }
+neqo-qpack = { path = "../neqo/neqo-qpack", version = "0.14.1" }
+neqo-crypto = { path = "../neqo/neqo-crypto", version = "0.14.1" }
+neqo-udp = { path = "../neqo/neqo-udp", version = "0.14.1" }
+
+[patch."https://github.com/beurdouche/mls-rs"]
+mls-rs = { path = "../mls-rs/mls-rs" }
+mls-rs-crypto-nss = { path = "../mls-rs/mls-rs-crypto-nss" }
+mls-rs-provider-sqlite = {path = "../mls-rs/mls-rs-provider-sqlite" }
+
+[patch."https://github.com/beurdouche/mls-platform-api"]
+mls-platform-api = { path = "../mls-platform-api" }
--- ./third_party/rust/suggest/Cargo.toml	2025-08-19 22:48:22.541759937 +0200
+++ ./third_party/rust/suggest/Cargo.toml	2025-08-19 22:49:37.512073314 +0200
@@ -61,12 +61,9 @@
 thiserror = "1"
 unicase = "2.6"
 unicode-normalization = "0.1"
-
-[dependencies.error-support]
-path = "../support/error"
-
-[dependencies.interrupt-support]
-path = "../support/interrupt"
+interrupt-support = "0.1.0"
+sql-support = "0.1.0"
+error-support = "0.1.0"
 
 [dependencies.remote_settings]
 path = "../remote_settings"
@@ -84,9 +81,6 @@
 version = "1"
 features = ["derive"]
 
-[dependencies.sql-support]
-path = "../support/sql"
-
 [dependencies.tempfile]
 version = "3.2.0"
 optional = true
@@ -102,7 +96,7 @@
 path = "../viaduct"
 
 [dependencies.viaduct-reqwest]
-path = "../support/viaduct-reqwest"
+path = "../../../application-services/components/support/viaduct-reqwest"
 optional = true
 
 [dev-dependencies]
@@ -111,10 +105,6 @@
 hex = "0.4"
 itertools = "0.14"
 
-[dev-dependencies.error-support]
-path = "../support/error"
-features = ["testing"]
-
 [dev-dependencies.rc_crypto]
 path = "../support/rc_crypto"
 
--- ./third_party/rust/search/Cargo.toml	2025-08-19 22:50:15.662403495 +0200
+++ ./third_party/rust/search/Cargo.toml	2025-08-19 22:51:00.264111808 +0200
@@ -32,12 +32,10 @@
 parking_lot = ">=0.11,<=0.12"
 serde_json = "1"
 thiserror = "1"
-
-[dependencies.error-support]
-path = "../support/error"
+error-support = "0.1.0"
 
 [dependencies.firefox-versioning]
-path = "../support/firefox-versioning"
+path = "../../../application-services/components/support/firefox-versioning"
 
 [dependencies.remote_settings]
 path = "../remote_settings"
@@ -55,7 +53,7 @@
 pretty_assertions = "0.6"
 
 [dev-dependencies.error-support]
-path = "../support/error"
+version = "0.1.0"
 features = ["testing"]
 
 [dev-dependencies.viaduct-reqwest]
--- ./third_party/rust/filter_adult/Cargo.toml	2025-08-19 22:51:27.129492547 +0200
+++ ./third_party/rust/filter_adult/Cargo.toml	2025-08-19 22:51:41.354331104 +0200
@@ -46,7 +46,7 @@
 optional = true
 
 [dependencies.error-support]
-path = "../support/error"
+path = "../../../application-services/components/support/error"
 
 [dependencies.uniffi]
 version = "0.29.0"
--- ./gfx/wgpu_bindings/Cargo.toml	2025-08-19 22:54:25.349216619 +0200
+++ ./gfx/wgpu_bindings/Cargo.toml	2025-08-19 22:55:10.989527541 +0200
@@ -16,8 +16,7 @@
 
 [dependencies.wgc]
 package = "wgpu-core"
-git = "https://github.com/gfx-rs/wgpu"
-rev = "b83c9cfd578837a6163d980130249c245a9c5f8a"
+path = "../../../wgpu/wgpu-core"
 # TODO: remove the replay feature on the next update containing https://github.com/gfx-rs/wgpu/pull/5182
 features = [
   "serde",
@@ -32,33 +31,28 @@
 # (We should consider also enabling "vulkan" for Vulkan Portability.)
 [target.'cfg(any(target_os = "macos", target_os = "ios"))'.dependencies.wgc]
 package = "wgpu-core"
-git = "https://github.com/gfx-rs/wgpu"
-rev = "b83c9cfd578837a6163d980130249c245a9c5f8a"
+path = "../../../wgpu/wgpu-core"
 features = ["metal"]
 
 # We want the wgpu-core Direct3D backends on Windows.
 [target.'cfg(windows)'.dependencies.wgc]
 package = "wgpu-core"
-git = "https://github.com/gfx-rs/wgpu"
-rev = "b83c9cfd578837a6163d980130249c245a9c5f8a"
+path = "../../../wgpu/wgpu-core"
 features = ["dx12"]
 
 # We want the wgpu-core Vulkan backend on Linux and Windows.
 [target.'cfg(any(windows, all(unix, not(any(target_os = "macos", target_os = "ios")))))'.dependencies.wgc]
 package = "wgpu-core"
-git = "https://github.com/gfx-rs/wgpu"
-rev = "b83c9cfd578837a6163d980130249c245a9c5f8a"
+path = "../../../wgpu/wgpu-core"
 features = ["vulkan"]
 
 [dependencies.wgt]
 package = "wgpu-types"
-git = "https://github.com/gfx-rs/wgpu"
-rev = "b83c9cfd578837a6163d980130249c245a9c5f8a"
+path = "../../../wgpu/wgpu-types"
 
 [dependencies.wgh]
 package = "wgpu-hal"
-git = "https://github.com/gfx-rs/wgpu"
-rev = "b83c9cfd578837a6163d980130249c245a9c5f8a"
+path = "../../../wgpu/wgpu-hal"
 features = ["device_lost_panic", "internal_error_panic"]
 
 [target.'cfg(windows)'.dependencies]
